mitodl/unified-ecommerce
# Adds third-party auth support to the app.

_This is intended to be reviewed and merged after https://github.com/mitodl/unified-ecommerce/pull/96 as that has the required infrastructure support. It's not strictly required for this, though._

### What are the relevant tickets?

https://github.com/mitodl/hq/issues/3166
https://github.com/mitodl/hq/issues/3158

### Description (What does it do?)

This adds:
- Authentication backend pieces for retrieving Keycloak user data
- Test routes for Keycloak integration
- Pulls in some remaining necessary libraries (fsm, reversion, oauth)

### How can this be tested?

Building and running the app should work. At this point, you will absolutely need a Keycloak instance set up and a realm configured. See the README for details. (The README won't be right until the dependent PR is merged.) 

There should be two new `meta` endpoints that you can use for testing: `traefik_test_request` and `authed_traefik_test_request`. These should be hit through Traefik and then you should be redirected through Keycloak for login, and then you should see keys and a user populate in the appropriate spots in the Django Admin.
